scripts/js/commands/checkSpelling.ts

```diff
@@ -0,0 +1,101 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { $ } from "zx";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { spellCheckDocument } from "cspell-lib";
+import { globby } from "globby";
+import pMap from "p-map";
+
+import { zxMain } from "../lib/zx.js";
+import { Pkg } from "../lib/api/Pkg.js";
+import { readMarkdown } from "../lib/markdownReader.js";
+
+interface Arguments {
+  [x: string]: unknown;
+  apis: boolean;
+  config: string;
+}
+
+function readArgs(): Arguments {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("apis", {
+      type: "boolean",
+      default: false,
+      description: "Check the API docs (currently failing)",
+    })
+    .option("config", {
+      type: "string",
+      default: "scripts/config/cspell/cSpell.json",
+      description: "Path to cSpell.json",
+    })
+    .parseSync();
+}
+
+zxMain(async () => {
+  const args = readArgs();
+  const cspellCmd = ["npx", "cspell", "--no-progress", "--config", args.config];
+
+  await $`${cspellCmd} docs/**/*.mdx !docs/api/**/*.mdx docs/api/migration-guides/*.mdx`;
+
+  if (args.apis) {
+    const apiFolders = Pkg.VALID_NAMES.map((api) => `docs/api/${api}/**/*.mdx`);
+    await $`${cspellCmd} ${apiFolders}`;
+  }
+
+  await checkAllNotebooks(args.config);
+});
+
+async function checkAllNotebooks(configPath: string): Promise<void> {
```

Comment:
Thoughts appreciated on how to make the logging more useful. Please take a look at the PR description for example output

I don't know if we want to unconditionally log that we're checking n Jupyter notebooks? Might be noisy.

Two ideas I just thought of:

* Add an emoji warning for each failure
* If it's the first failure encountered, use `\n\n` so it stands out from the logs above it.

Reply:
I think a "❌" emoji usually works well